pkg/binding/selectors.go

```diff
@@ -93,8 +96,8 @@ func (c *Controller) updateResolutions(ctx context.Context, objIdentifier util.O
 				// this case can occur if a bindingpolicy resolution was deleted AFTER
 				// starting this iteration and BEFORE getting to the NoteObject function,
 				// which occurs if a bindingpolicy was deleted in this time-window.
```

Comment:
FYI: Now the window is even smaller, it opens after the new check above for `c.bindingPolicyResolver.ResolutionExists`.